### PR TITLE
Add fogColor uniform

### PR DESCRIPTION
**Pull request by IMS212**
This will likely fix fog in the nether in multiple shaders, along with biome fog.

### CODE REVIEW

**Comment by coderbot16** [comment]
This method works with vanilla but might mess up in the context of mods. https://github.com/CaffeineMC/sodium-fabric/blob/1.16.x/next/src/main/java/me/jellysquid/mods/sodium/client/render/chunk/shader/ChunkFogMode.java is an example of a way that would work in all cases, though it is not 100% optimal because it could introduce pipeline stalls (which should be noted in a TODO) — as such, calls to that method should be minimized as much as possible. And, of course, in 1.17, it's possible to retrieve the fog mode directly from RenderSystem, which would make the issues with that previous approach sort of moot.

**Comment by IMS212** [comment]
I've redone this PR to use a mixin into BackgroundRenderer instead of a manual calculation; this should be identical to vanilla's fog color now.

File: gradle.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ org.gradle.jvmargs=-Xmx2G
 # Fabric Properties
 # check these on https://fabricmc.net/use
 minecraft_version=1.16.5
-yarn_mappings=1.16.5+build.5
+yarn_mappings=1.16.5+build.6
```
Comment (FoundationGames) [inline gradle.properties:6]: Just wondering why this is necessary, but everything else looks great!

**Review by FoundationGames — APPROVED** [review]

File: gradle.properties (hunk 1 above)
Comment (IMS212) [reply gradle.properties:6]: Ah, that’s just a leftover from my Linux setup. For some reason build 5 gives an unexplainable error when trying to use runClient on my Arch setup, and build 6 doesn’t have any conflicts anyway so I don’t see any reason not to.

File: src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java [hunk 2]
```diff
@@ -21,8 +21,10 @@
 import net.minecraft.item.BlockItem;
 import net.minecraft.item.ItemStack;
 import net.minecraft.tag.FluidTags;
+import net.minecraft.util.CubicSampler;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java:24]: This seems unused.

File: src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java [hunk 3]
```diff
@@ -21,8 +21,10 @@
...
 import net.minecraft.util.Hand;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java:27]: Same here

File: src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java [hunk 4]
```diff
@@ -32,6 +34,7 @@
 
 public final class CommonUniforms {
 	private static final MinecraftClient client = MinecraftClient.getInstance();
+	private static Vec3d fogColor;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java:37]: This should probably be a member of CapturedRenderingState for the moment.

**Review by coderbot16 — CHANGES_REQUESTED** [review]
This should be a good way to go about things until I figure out how to directly pull / save fog data from GlStateManager. I'd like a few things addressed first though.

**Comment by coderbot16** [comment]
Apparently this causes a NullPointerException to occur on Complementary since the fog color isn't set early in the frame

**Comment by IMS212** [comment]
All these issues should be solved now.

**Review by coderbot16 — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**